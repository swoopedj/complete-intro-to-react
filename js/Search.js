import React from 'react'
import preload from '../public/data.json'
import ShowCard from './ShowCard'

const Search = React.createClass({
  getInitialState () {
    return {
      searchTerm: 'this is the default string'
    }
  },
  handleSearchTermChange (event) {
    this.setState({searchTerm: event.target.value})
  },
  render () {
    return (
      <div className='search'>
        <header>
          <h1>svideo</h1>
          <input onChange={this.handleSearchTermChange} value={this.state.searchTerm} type='text' placeholder='Search' />
        </header>
        <div>
          {preload.shows
            .filter((show) => {
              return `${show.title} ${show.title}`
                .toUpperCase().indexOf(this.state.searchTerm
                .toUpperCase()) >= 0
            })
            .map((show) => {
            return (
              <ShowCard key={show.imdbID} {...show} />
            )
          })}
        </div>
      </div>
    )
  }
})

export default Search
