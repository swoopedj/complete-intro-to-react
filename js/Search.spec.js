import React from 'react'
import Search from './search'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

test('Search snapshot test', () => {
  const component = shallow(<Search />)
  const tree = shallowToJson
  expect(tree).toMatchSnapshot()
})