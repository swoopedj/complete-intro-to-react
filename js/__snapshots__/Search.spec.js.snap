exports[`test Search snapshot test 1`] = `[Function shallowToJson]`;
